java/gazelle/javaconfig/codegen/codegen.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"log"
+	"os"
+	"reflect"
+	"strings"
+	"unicode"
+
+	bzl "github.com/bazelbuild/buildtools/build"
+)
+
+func main() {
+	if len(os.Args) != 3 {
+		log.Fatalf("Want 2 args, got: %d: %v", len(os.Args)-1, os.Args[1:])
+	}
+	inputPath := os.Args[1]
+	outputPath := os.Args[2]
+
+	contents, err := os.ReadFile(inputPath)
+	if err != nil {
+		log.Fatalf("Failed to read %v: %v", inputPath, err)
+	}
+	file, err := bzl.ParseBzl(inputPath, contents)
+	if err != nil {
+		log.Fatalf("Failed to parse %v: %v", inputPath, err)
+	}
+
+	var defaultTestSuffixes []string
+
+	for _, expr := range file.Stmt {
+		if assign, ok := expr.(*bzl.AssignExpr); ok {
+			if ident, ok := assign.LHS.(*bzl.Ident); ok {
+				if ident.Name == "DEFAULT_TEST_SUFFIXES" {
+					if rhs, ok := assign.RHS.(*bzl.ListExpr); ok {
+						for _, expr := range rhs.List {
+							if lit, ok := expr.(*bzl.StringExpr); ok {
+								for _, r := range []rune(lit.Value) {
+									if !unicode.IsLetter(r) && !unicode.IsNumber(r) && r != '.' {
+										// We're going to convert these strings to regexes - this is a simple check to make sure we can do so safely.
+										log.Fatalf("DEFAULT_TEST_SUFFIXES may only contain alphanumeric characters or . but got %q", lit.Value)
+									}
+								}
+								defaultTestSuffixes = append(defaultTestSuffixes, lit.Value)
```

Comment:
Should we fail if the list is re-defined instead of doing an append?

In the input is:

```starlark
DEFAULT_TEST_SUFFIXES = ["Test.java"]
DEFAULT_TEST_SUFFIXES = ["Test.java"]
DEFAULT_TEST_SUFFIXES = ["Test.java"]
```

The current output is:

```go
package javaconfig

import "regexp"

var defaultTestRegexps = []*regexp.Regexp{
regexp.MustCompile(".*Test\\.java$"),
regexp.MustCompile(".*Test\\.java$"),
regexp.MustCompile(".*Test\\.java$"),
}
```

Reply:
Absolutely! Done.